File: stack/java/stack.java
public class Stack{
    private int[] arr;
    private int top;
    private int capacity;

    public Stack(int initialCapacity) {
        arr = new int[initialCapacity];
        top = -1;
        capacity = initialCapacity;
    }

    public Stack(){
        this(5);
    }

    public boolean isEmpty(){
        return top == -1;
    }

    public boolean isFull(){
        return top == capacity - 1;
    }

    public void push(int data){
        if(isFull()){
            System.out.println("Stack overflow");
            return;
        }
        arr[++top] = data;
    }

    public int pop(){
        if(isEmpty()){
            System.out.println("Stack underflow");
            throw new IllegalStateException("Stack underflow");
        }
        return arr[top--];
    }

    public int peek(){
        if(isEmpty()){
            System.out.println("Stack underflow");
            throw new IllegalStateException("Stack underflow");
        }
        return arr[top];
    }


    public static void main(String[] args)
    {
        Stack myStack = new Stack();

        myStack.push(10);
        myStack.push(20);
        myStack.push(30);
        myStack.push(40);
        myStack.push(50);

        System.out.println("Is the stack full? " + (myStack.isFull() ? "Yes" : "No"));
        System.out.println("Top element: " + myStack.peek());
        System.out.println("Popped element: " + myStack.pop());
        System.out.println("Popped element: " + myStack.pop());

        System.out.println("Is the stack full? " + (myStack.isFull() ? "Yes" : "No"));
        System.out.println("Is the stack empty? " + (myStack.isEmpty() ? "Yes" : "No"));
    }

}

File: stack/c/stack.c
#include <stdio.h>
#include <stdlib.h>
#define MAX 4
int stack_arr[MAX];
int top = -1;

void push(int data)
{
	if (top == MAX - 1)
	{
		printf("Stack overflow\n");
		return;
	}
	top = top + 1;
	stack_arr[top] = data;
}
int pop()
{
	if (top == -1)
	{
		printf("Stack underflow\n");
		exit(1); // abnormal termination of program
	}
	int value;
	value = stack_arr[top];
	top = top - 1;
	return value;
}
int topElement()
{
	if (top == -1)
	{
		printf("Empty stack");
		return 0;
	}
	printf("%d\n", stack_arr[top]);
	return stack_arr[top];
}
void print()
{
	if (top == -1)
	{
		printf("Stack underflow\n");
		return;
	}
	for (int i = top; i >= 0; i--)
		printf("%d ", stack_arr[i]);

	printf("\n");
}

int main()
{
	int data;
	push(1);
	topElement();
	push(2);
	topElement();
	push(3);
	topElement();
	print();
	push(4);
	print();
	pop();
	print();
	pop();
	print();
	return 0;
}
